tests/test_hd44780.py

```diff
@@ -0,0 +1,187 @@
+
+    serial.data.assert_has_calls(calls)
+
+
+def test_init_8bitmode():
+    """
+    Test initialization of display using 4 bit mode
+    """
+    serial._bitmode = 8
+    hd44780(serial)
+
+    to_8 = \
+        [call(0x30)] * 3
+
+    fs = [FUNCTIONSET | DL8 | LINES2]
+
+    calls = \
+        to_8 + \
+        [call(*fs)] + \
+        [call(*[DISPLAYOFF])] + \
+        [call(*[ENTRY])] + \
+        [call(*[DISPLAYON])] + \
+        [call(*[DDRAMADDR])] + \
+        [call(*[DDRAMADDR | LINES[1]])] + \
+        [call(*[CLEAR])]
+
+    serial.command.assert_has_calls(calls)
+
+    # Data to clear the screen
+    calls = \
+        [call([0x20] * 16)] + \
+        [call([0x20] * 16)]
+
+    serial.data.assert_has_calls(calls)
+
+
+def test_display():
+    """
+    Test the display of a line of text containing and a rectangle to verify
+    auto-create feature
+    """
+    device = hd44780(serial, bitmode=8)
+    serial.reset_mock()
+
+    # Use canvas to create a screen worth of data
+    with canvas(device) as drw:
+        # Include unprintable character to show it gets ignored
+        size = device.font.getsize('This is a test\uFFFF')
+        drw.text(((80 - size[0]) // 2, 0), 'This is a test\uFFFF', font=device.font, fill='white')
+        drw.rectangle((10, 10, 69, 14), fill='black', outline='white')
+        drw.rectangle((10, 10, 49, 14), fill='white', outline='white')
+
+    # Send DDRAMADDR and ascii for the line of text
+    line1 = [call.command(0x81)] + \
+        [call.data([0x54, 0x68, 0x69, 0x73, 0x20, 0x69, 0x73, 0x20, 0x61, 0x20, 0x74, 0x65, 0x73, 0x74])]
+
+    # Create custom characters for the scrollbar
+    custom = [call.command(0x40), call.data([0x00, 0x00, 0x1f, 0x1f, 0x1f, 0x1f, 0x1f, 0x00])] + \
+        [call.command(0x48), call.data([0x00, 0x00, 0x1f, 0x00, 0x00, 0x00, 0x1f, 0x00])] + \
+        [call.command(0x50), call.data([0x00, 0x00, 0x1f, 0x01, 0x01, 0x01, 0x1f, 0x00])]
+
+    # Print the resulting custom characters to form the image of the scrollbar
+    line2 = [call.command(0xc1), call.data([0x20, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x01, 0x01, 0x02, 0x20])]
+
+    serial.assert_has_calls(line1 + custom + line2)
+
+
+def test_custom_full():
+    """
+    Test of behavior when auto-create feature runs out of custom character
+    space
+    """
+    device = hd44780(serial, bitmode=8)
+
+    # Consume 8 special character positions
+    img = Image.new('1', (80, 16), 0)
+    drw = ImageDraw.Draw(img)
+    for i in range(8):
+        drw.rectangle((i * 5, 0, (i + 1) * 5, i), fill='white', outline='white')
+    device.display(img)
+
+    serial.reset_mock()
+
+    # Consume one more (on the last char position on screen)
+    drw.line((75, 8, 79, 15), fill='white')
+    device.display(img)
+
+    serial.assert_has_calls([call.command(0xcf), call.data([0x5f])])
+
+
+def test_get_font():
+    """
+    Verify get font capability by requesting two fonts and printing a single
```

Comment:
No need to use word "Verify". I'd omit it, tests are always verifying.

Reply:
I've eliminated the use of the word verify.